File: Problem-Of-The-Day/2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> neg, pos;
        int i = 0;
        for (int n : nums) {
            if (n > 0)
                pos.push_back(n);
            else
                neg.push_back(n);
        }
        while (2 * i < nums.size()) {
            nums[2 * i] = pos[i];
            nums[2 * i + 1] = neg[i];
            ++i;
        }
        return nums;
    }
};

File: Problem-Of-The-Day/2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        neg, pos, i = [], [], 0
        for n in nums:
            if n > 0:
                pos.append(n)
            else:
                neg.append(n)
        while 2 * i < len(nums):
            nums[2 * i] = pos[i]
            nums[2 * i + 1] = neg[i]
            i += 1
        return nums
